src/EditorFeatures/Test/AbstractCommandHandlerTestState.cs

```diff
@@ -64,25 +73,27 @@ internal abstract class AbstractCommandHandlerTestState : IDisposable
 
             if (cursorDocument.AnnotatedSpans.TryGetValue("Selection", out selectionSpanList))
             {
-                var span = selectionSpanList.First();
+                var firstSpan = selectionSpanList.First();
+                var lastSpan = selectionSpanList.Last();
                 var cursorPosition = cursorDocument.CursorPosition.Value;
 
-                Assert.True(cursorPosition == span.Start || cursorPosition == span.Start + span.Length,
+                Assert.True(cursorPosition == firstSpan.Start || cursorPosition == lastSpan.End,
```

Comment:
This doesn't handle box selection correctly, there might be multiple spans and the cursor could be located at the last span.